warning: not reporting region error due to -Znll
  --> $DIR/propagate-approximated-to-static.rs:44:9
   |
44 |         demand_y(x, y, x.get()) //~ WARNING not reporting region error due to -Znll
   |         ^^^^^^^^^^^^^^^^^^^^^^^

note: External requirements
  --> $DIR/propagate-approximated-to-static.rs:40:47
   |
40 |       establish_relationships(&cell_a, &cell_b, |_outlives, x, y| {
   |  _______________________________________________^
41 | |         //~^ ERROR free region `'_#1r` does not outlive free region `ReStatic`
42 | |
43 | |         // Only works if 'x: 'y:
44 | |         demand_y(x, y, x.get()) //~ WARNING not reporting region error due to -Znll
45 | |     });
   | |_____^
   |
   = note: defining type: [closure@$DIR/propagate-approximated-to-static.rs:40:47: 45:6<i16, for<'r, 's, 't0, 't1, 't2> extern "rust-call" fn((&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 'r(7714))) std::cell::Cell<&'_#1r &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 's(7715))) u32>, &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 't0(7716))) std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 's(7715))) u32>, &ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 't1(9540))) std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(0:0), 't2(9541))) u32>))>]
   = note: number of external vids: 2
   = note: where '_#1r: '_#0r

error: free region `'_#1r` does not outlive free region `ReStatic`
  --> $DIR/propagate-approximated-to-static.rs:40:47
   |
40 |       establish_relationships(&cell_a, &cell_b, |_outlives, x, y| {
   |  _______________________________________________^
41 | |         //~^ ERROR free region `'_#1r` does not outlive free region `ReStatic`
42 | |
43 | |         // Only works if 'x: 'y:
44 | |         demand_y(x, y, x.get()) //~ WARNING not reporting region error due to -Znll
45 | |     });
   | |_____^

note: No external requirements
  --> $DIR/propagate-approximated-to-static.rs:39:1
   |
39 | / fn supply<'a, 'b>(cell_a: Cell<&'a u32>, cell_b: Cell<&'b u32>) {
40 | |     establish_relationships(&cell_a, &cell_b, |_outlives, x, y| {
41 | |         //~^ ERROR free region `'_#1r` does not outlive free region `ReStatic`
42 | |
...  |
45 | |     });
46 | | }
   | |_^
   |
   = note: defining type: for<'a, 'b> fn(std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(1:16), 'a(90))) u32>, std::cell::Cell<&ReLateBound(DebruijnIndex { depth: 1 }, BrNamed(crate0:DefIndex(1:17), 'b(92))) u32>) {supply}

error: aborting due to previous error

